scripts/scil_assign_custom_color_to_tractogram.py

```diff
@@ -197,27 +199,23 @@ def main():
         elif args.use_dpp:
             data = np.squeeze(sft.data_per_point[args.use_dpp]._data)
         elif args.load_dps:
-            data = np.squeeze(load_matrix_in_any_format(args.load_dps))
-            if len(data) != len(sft):
-                parser.error('Wrong dps size!')
+            sft, _ = load_dps_files_as_dps(parser, [args.load_dps], sft,
+                                           keys=['from_dps'])
+            data = sft.data_per_streamline['from_dps']
         else:  # args.load_dpp
-            data = np.squeeze(load_matrix_in_any_format(args.load_dpp))
-            if len(data) != len(sft.streamlines._data):
-                parser.error('Wrong dpp size!')
+            sft, _ = load_dpp_files_as_dpp(parser, [args.load_dps], sft,
+                                           keys=['from_dpp'])
+            data = sft.data_per_point['from_dpp']
         values, lbound, ubound = clip_and_normalize_data_for_cmap(args, data)
     elif args.from_anatomy:
         data = nib.load(args.from_anatomy).get_fdata()
         data, lbound, ubound = clip_and_normalize_data_for_cmap(args, data)
-
-        sft.to_vox()
-        values = map_coordinates(data, sft.streamlines._data.T, order=0)
-        sft.to_rasmm()
+        sft = add_map_values_as_dpp(sft, [data], ['from_anatomy'])
+        values = sft.data_per_point['from_anatomy']
     elif args.along_profile:
-        values, lbound, ubound = get_color_streamlines_along_length(
-            sft, args)
+        values, lbound, ubound = get_color_streamlines_along_length(sft, args)
```

Comment:
I would argue that passing "args" to a function is really not much better than having everything in the main. Having "args" as an argument prevents any other piece of code from calling it, which is really bad practice. Would it be possible to pass actual parameters instead of the whole argparser ? 

For example, a use case would be: In TractOracle, I would like to score streamlines and assign them a color directly. However, I can't use this function without some complicated mocking.